R/IntakeDetailedProtein.R

```diff
@@ -0,0 +1,49 @@
+#' @title IntakeDetailedProtein
+#' @description Calculates detailed per-capita (protein in grams) intake from magpie results at regional level
+#' @export
+#' @param gdx GDX file
+#' @param file a file name the output should be written to using write.magpie
+#' @return Product disaggregated Protein intake as MAgPIE object at regional level (unit: grams/cap/day)
+#' @author Vartika Singh, Isabelle Weindl
+#' @importFrom gdx readGDX
+#' @importFrom magclass dimSums
+#' @examples
+#' 
+#'   \dontrun{
+#'     x <- IntakeDetailedProtein(gdx)
+#'   }
+#' 
+
+IntakeDetailedProtein <- function(gdx,file=NULL){
```

Comment:
I'm not so familiar with this calculation, but by looking at `IntakeDetailed.R` function, as a parent function, I wonder if it would be also useful to add flexibility in here by keeping some (or all) function arguments that exit in `IntakeDetailed.R`? E.g. `level`, `product_aggr`, and girdded stuff I find useful to consider...

Reply:
alternatively, have you considered added an argument in `IntakeDetailed.R` that can calculate the protein directly from there?

Reply:
Adding all arguments back into the new function would make this function more complex. Same holds true for adding the unit conversion into IntakeDetailed, especially due to the different treatment for the cases for magpie_input.

But introducing "level, product_aggr, and gridded stuff" is a good idea.

Adding level and gridded stuff is possible via:
 out<-gdxAggregate(gdx = gdx,x = intake_scen,weight = 'population',to = level,absolute = FALSE,dir = dir)

And product_aggr=T can be realized via dimSums in the end of the function.

Reply:
But maybe add in the description that this function is only able to calculate the protein content of food intake estimated by the food demand model, and not of food intake that is prescribed in the case of exogenous diet scenarios (e.g. EAT Lancet diets).

Reply:
Thank you both. I have now majorly updated the function to obtain information from the IntakeDetailed function and apply the relevant changes to calculate protein intake information for all types of diets. Also added in the options for production disaggregation, gridded outputs and levels.